### ISSUE TITLE
Mention explicitly how WSL 2 distros run as containers under a single kernel

### ISSUE THREAD

**Issue by cerebrate** [MicrosoftDocs/WSL#508]
This particular detail of the WSL architecture makes it behave in ways different from a "generic" Linux installation (e.g., kernel threads do not show up in a process listing, et. sim.), and these ways are unexpected if you don't know what's going on under the hood. And hence it would be useful to have that in the docs.

(Initially reported at https://github.com/microsoft/WSL/issues/4622 .)

**Comment by craigloewen-msft** [comment]
Agreed! I aim to add a 'How WSL2' works doc as we approach general release and will make sure this detail is added in there.

**Comment by mattwojo** [comment]
@craigloewen-msft -- is this adequetly covered here? https://docs.microsoft.com/en-us/windows/wsl/compare-versions

If so, please close. :) ~Matt

**Comment by craigloewen-msft** [comment]
Not completely. Really the answer here would be creating a full 'How does WSL 2' work doc that basically explains this diagram:
https://www.youtube.com/watch?v=lwhMThePdIo&t=1800s

**Comment by mattwojo** [comment]
Sounds good. Drafted a WSL 2 architecture explainer doc -- could you help me fill it in with the correct info? 
https://github.com/MicrosoftDocs/WSL/blob/mattw-wsl2-explainer/WSL/wsl2-architecture.md

**Comment by cerebrate** [comment]
It would probably be useful to include a quick reference list of things that aren't covered by the namespacing that separates distros.

If I'm reading lsns right, the distros have their own cgroup/user/net/uts/ipc/mnt/pid namespaces, but not time namespaces (understandably, since those have only been around since kernel 5.6 and the WSL kernel hasn't reached that yet), so the system clocks mentioned in time_namespaces(7) are shared between distros. Admittedly, these are nonsettable clocks outside any clock offsets you create while creating the namespace, so I can't think of a reason offhand why you'd want to do that for WSL distros anyway.

On the other hand, RTC time is shared between distros (with a potential relevant use case for changing time to run tests). Other things that come to mind are loaded kernel modules, the kernel keyring, kernel security policies, and probably a variety of other things I don't know Linux well enough to know of...

**Comment by vbrozik** [comment]
> Not completely. Really the answer here would be creating a full 'How does WSL 2' work doc that basically explains this diagram: https://www.youtube.com/watch?v=lwhMThePdIo&t=1800s

The particular part about the WSL2 instances (Microsoft calls them "distributions") sharing a single VM, single Linux kernel instance and using namespaces is mentioned here:
https://youtu.be/lwhMThePdIo?t=2436

It would be really useful to have at lease a rough description of these:
- how is the kernel booted
- how are the kernel parameters set
- how is the initial filesystem mounted
- how does the WSL's `init` work
- how are the namespaces created
- how are the "distributions" mounted and executed (do they use the same `init` binary?)
